clients/githubrepo/branches.go

```diff
@@ -476,23 +493,34 @@ nextRule:
 	return ret, nil
 }
 
+func initializedBoolRef(value bool) *bool {
+	v := new(bool)
+	*v = value
+	return v
+}
+
 func applyRepoRules(branchRef *clients.BranchRef, rules []*repoRuleSet) {
-	falseVal := false
-	trueVal := true
 	for _, r := range rules {
-		adminEnforced := len(r.BypassActors.Nodes) == 0
+		// Init values of base checkbox as if they're unchecked
 		translated := clients.BranchProtectionRule{
-			EnforceAdmins: &adminEnforced,
+			AllowDeletions:       initializedBoolRef(true),
+			AllowForcePushes:     initializedBoolRef(true),
+			RequireLinearHistory: initializedBoolRef(false),
 		}
 
+		translated.EnforceAdmins = initializedBoolRef(len(r.BypassActors.Nodes) == 0)
```

Comment:
Good point. There are 2 main changes that I made around this part of the code:
1. Creating and using this initializedBoolRef() function to substitute the `&falseVal` values. I made this just because I thought this could not be really safe because if the values on `falseVal` or `trueVal` were changed somewhere, it would led to bugs difficult to track.
This is completely outside the scope of the PR and I should have made this on a different PR, sorry.

2. Initialize the fields with the value they should get if the checkbox remains unchecked. I made this because the logic of this function is only changing the values if the checkbox gets checked, and we often were getting `nil` values when the user actually made the decision to not check that field. I don't remember if this was already leading to any bugs, but I do believe it eventually would, because we often interpret `nil` as "we don't have the data". I've created [this test ](https://github.com/diogoteles08/scorecard/blob/feat/branch-protection-recognize-rule-changes-only-through-pr/clients/githubrepo/branches_test.go#L194) to cover this change.
This is also arguably outside the scope of this PR, but makes a bit more sense since we have discussed a lot the meanings of `nil` pointers, etc.